packages/wrangler/e2e/wrangler-mixed-mode-resources.test.ts

```diff
@@ -0,0 +1,304 @@
+			const name = await helper.r2(false);
+			await helper.run(
+				`wrangler r2 object put --remote ${name}/test-mixed-mode-key --file test.txt`
+			);
+			onTestFinished(async () => {
+				await helper.run(
+					`wrangler r2 object delete --remote ${name}/test-mixed-mode-key`
+				);
+			});
+			return { name };
+		},
+		generateWranglerConfig: ({ name: bucketName }) => ({
+			name: "mixed-mode-r2-test",
+			main: "r2.js",
+			compatibility_date: "2025-01-01",
+			r2_buckets: [
+				{
+					binding: "R2_BINDING",
+					bucket_name: bucketName,
+					remote: true,
+				},
+			],
+		}),
+		expectedResponseMatch: "The pre-existing value is: existing-value",
+	},
+	{
+		name: "D1",
+		scriptPath: "d1.js",
+		setup: async (helper) => {
+			await helper.seed({
+				"schema.sql": dedent`
+          CREATE TABLE entries (key TEXT PRIMARY KEY, value TEXT);
+          INSERT INTO entries (key, value) VALUES ('test-mixed-mode-key', 'existing-value');
+        `,
+			});
+			const { id, name } = await helper.d1(false);
+			await helper.run(
+				`wrangler d1 execute --remote ${name} --file schema.sql`
+			);
+			return { id, name };
+		},
+		generateWranglerConfig: ({ id, name }) => ({
+			name: "mixed-mode-d1-test",
+			main: "d1.js",
+			compatibility_date: "2025-01-01",
+			d1_databases: [
+				{
+					binding: "DB",
+					database_id: id,
+					database_name: name,
+					remote: true,
+				},
+			],
+		}),
+		expectedResponseMatch: "existing-value",
+	},
+];
+
+describe("Wrangler Mixed Mode E2E Tests", () => {
+	describe.each(testCases)("$name", (testCase) => {
+		let helper: WranglerE2ETestHelper;
+
+		beforeEach(() => {
+			helper = new WranglerE2ETestHelper();
+		});
+
+		it("works with mixed mode enabled", async () => {
+			await helper.seed(
+				path.resolve(__dirname, "./seed-files/mixed-mode-workers")
+			);
+
+			await writeWranglerConfig(testCase, helper);
+
+			const worker = helper.runLongLived("wrangler dev --x-mixed-mode");
+
+			const { url } = await worker.waitForReady();
+
+			const response = await fetchText(url);
+			expect(response).toMatch(testCase.expectedResponseMatch);
+		});
+
+		it.skipIf(testCase.worksWithoutMixedMode)(
+			"fails when mixed mode is disabled",
+			async () => {
+				await helper.seed(
+					path.resolve(__dirname, "./seed-files/mixed-mode-workers")
+				);
+
+				await writeWranglerConfig(testCase, helper);
+
+				const worker = helper.runLongLived("wrangler dev");
+
+				const { url } = await worker.waitForReady();
+
+				const response = await fetchText(url);
+				expect(response).not.toMatch(testCase.expectedResponseMatch);
+			}
+		);
+	});
+
+	describe.sequential("Sequential mixed mode tests with worker reloads", () => {
+		let worker: WranglerLongLivedCommand;
+		let helper: WranglerE2ETestHelper;
+
+		let url: string;
+
+		beforeAll(async () => {
+			helper = new WranglerE2ETestHelper();
+			await helper.seed(
+				path.resolve(__dirname, "./seed-files/mixed-mode-workers")
```

Comment:
Should this use a different path to the `describe` block above so that the seed files are isolated?